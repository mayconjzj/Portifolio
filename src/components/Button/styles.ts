import tw from 'tailwind-styled-components';

export const Button = tw.button`
  text-white
  font-medium
  border-2
  border-blue-500
  rounded-lg
  min-w-[150px]
  py-2
  px-4
  transition-colors
  delay-150
  hover:bg-blue-500
`;
