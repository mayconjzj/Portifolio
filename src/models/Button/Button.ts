import { ButtonHTMLAttributes } from 'react';

export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  children: React.ReactNode;
  className?: string;
};
