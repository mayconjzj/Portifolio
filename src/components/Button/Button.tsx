import * as S from './styles';

export const Button = ({
  children,
  ...rest
}: {
  children: React.ReactNode;
}) => {
  return <S.Button {...rest}>{children}</S.Button>;
};
