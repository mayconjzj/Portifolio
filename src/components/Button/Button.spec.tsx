import { render } from '@testing-library/react';

import { Button } from './Button';

describe('Button', () => {
  it('should render the button', () => {
    const { asFragment } = render(<Button>TESt</Button>);
    expect(asFragment()).toBeTruthy();

    expect(asFragment()).toMatchSnapshot();
  });
});
