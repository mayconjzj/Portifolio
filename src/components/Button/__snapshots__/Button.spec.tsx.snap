// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button 1`] = `
<div>
  <button
    class="text-white font-medium border-2 border-blue-500 rounded-lg min-w-[150px] py-2 px-4 transition-colors duration-300 hover:bg-blue-500"
  >
    Next test
  </button>
</div>
`;
